internal/resource/factory.go

```diff
@@ -39,7 +39,7 @@ func WithConfig(manager Manager, config *spec.Config) Manager {
 }
 
 // getManager returns the resource manager depending on the system configuration.
-func getManager() Manager {
+func getManager(config *spec.Config) Manager {
```

Comment:
See https://github.com/NVIDIA/k8s-device-plugin/pull/673 where I reworked this to use go-nvlib functionality that I'm trying to add.

If we add a `mode` flag, I would recommend  only passing the `mode` here and then making a construction decision based on that.

Reply:
@elezar This makes sense. How do you want to handle this? Since the go-nvlib PR is still open do you want to update the info interface or should I wait for it to be merged?

Reply:
I think the addition of the mode flag is separate from that PR. We can already add the config option and command line flags here and then ensure that it is plumbed through. I would suggest.

1. Add a new PR that adds the mode flag / config option.
2. Rebase these changes to include the mode flag commits, adding support for a `vfio` mode.

Once the go-nvlib changes and #673 are merged, we can update the checks for the existing modes here.

As a follow-up we can pull the `vfio` detection into `go-nvlib`, but I don't see that as a blocker for this PR.